core/src/main/java/com/linecorp/armeria/server/http/healthcheck/ManagedHttpHealthCheckService.java

```diff
@@ -0,0 +1,111 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.healthcheck;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import com.google.common.base.Ascii;
+
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.http.AggregatedHttpMessage;
+import com.linecorp.armeria.common.http.HttpData;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * A {@link HttpHealthCheckService} which allows overriding its status via a POST request.
+ * <p>
+ * <h2>Example</h2>
+ * <pre>{@code
+ * Server server = new ServerBuilder().serviceUnder("/health", new ManagedHttpHealthCheckService())
+ *                                    .build()
+ * }</pre>
+ * Default config uses content on or off
+ * You can also use your own path matching.
+ * </p>
+ * <pre>{@code
+ * ...
+ * Server server = new ServerBuilder().serviceUnder("health", new ManagedHttpHealthCheckService() {
+ *         {@literal @}Override
+ *         public CompletableFuture&lt;Optional&lt;Boolean&gt;&gt; mode(HttpRequest req) {
+ *             return CompletableFuture.completedFuture(Optional.empty());
+ *         }
+ *     }).build())
+ * ...
+ * }</pre>
+ */
+public class ManagedHttpHealthCheckService extends HttpHealthCheckService {
+    private static final AggregatedHttpMessage TURN_ON_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, HttpData.ofUtf8("Set healthy."));
+    private static final AggregatedHttpMessage TURN_OFF_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, HttpData.ofAscii("Set unhealthy."));
+
+    @Override
+    protected void doPut(ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res) throws Exception {
+        updateHealthStatus(ctx, req).thenAccept(res::respond)
+                                    .exceptionally(cause -> {
+                                        res.close(cause);
+                                        return null;
+                                    });
+    }
+
+    /**
+     * Updates health status using the specified {@link HttpRequest}.
+     */
+    private CompletableFuture<AggregatedHttpMessage> updateHealthStatus(
+            ServiceRequestContext ctx, HttpRequest req) {
+        return mode(ctx, req).thenApply(mode -> {
+            if (!mode.isPresent()) {
+                return newResponse(ctx);
```

Comment:
I think we need to return an error status to the client such as `500 Internal Server Error` because `!mode.isPresent()` means the client sent a wrong request. In this case, the server may return `200 Ok` if the server is healthy. But this provides a wrong information to the client, so the client may think the request has succeeded.